pkg/cmd/cli/cmd/deploy.go

```diff
@@ -342,3 +374,109 @@ func (o *DeployOptions) reenableTriggers(config *deployapi.DeploymentConfig, out
 	fmt.Fprintf(out, "enabled image triggers: %s\n", strings.Join(enabled, ","))
 	return nil
 }
+
+// DeployLogOptions holds all the options for the `deploy --log` command
+type DeployLogOptions struct {
+	out      io.Writer
+	osClient client.Interface
+
+	name      string
+	namespace string
+
+	options deployapi.DeploymentLogOptions
+	version string
+}
+
+// Complete the DeployLogOptions
+func (o *DeployLogOptions) Complete(f *clientcmd.Factory, args []string, out io.Writer) error {
+	var err error
+
+	o.osClient, _, err = f.Clients()
+	if err != nil {
+		return err
+	}
+	o.namespace, _, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	o.out = out
+
+	if len(args) > 0 {
+		params := strings.Split(args[0], "/")
+		switch len(params) {
+		case 1:
+			o.name = params[0]
+		case 2:
+			// dc/name format
+			o.name = params[1]
+		default:
+			return fmt.Errorf("invalid resource type/name argument: %s", args[0])
+		}
+	}
+
+	if len(o.version) > 0 {
+		v64, err := strconv.ParseInt(o.version, 10, 0)
+		if err != nil {
+			return err
+		}
+		o.options.Version = intp(v64)
+	}
+
+	// Hardcode follow for now. Ideally oc deploy should have subcommands
+	o.options.Follow = true
+
+	return nil
+}
+
+func intp(n int64) *int {
+	num := new(int)
+	*num = int(n)
+	return num
+}
+
+// Validate all the necessary deploy log options
+func (o *DeployLogOptions) Validate(args []string) error {
+	if len(args) == 0 || len(args[0]) == 0 {
+		return errors.New("a DeploymentConfig name is required.")
+	}
+	if len(args) > 1 {
+		return errors.New("only one DeploymentConfig name is supported as argument.")
+	}
+	return nil
+}
+
+// RunDeployLog sets up the client for deploy --logs and executes the command
+func (o *DeployLogOptions) RunDeployLog() error {
+	return o.logs(o.namespace, o.name, o.options, o.out)
+}
+
+// logs returns the logs for the latest deployment
+func (o *DeployLogOptions) logs(namespace, name string, opts deployapi.DeploymentLogOptions, out io.Writer) error {
+	readCloser, err := o.osClient.DeploymentLogs(namespace).Get(name, opts).Stream()
+	if err != nil {
+		return err
+	}
+	defer readCloser.Close()
+
+	written, err := io.Copy(out, readCloser)
+	if err != nil {
+		return err
+	}
+
+	// If nothing is written, it means there are no logs returned. Normally
+	// in two cases we will get no logs: either if we don't want to wait
+	// for the deployer pod to be created (NoWait = true) or if the deployer
+	// pod has been deleted (successful deployment)
+	// TODO: In case of manual pod deletion or deployment pruning, this will
```

Comment:
Added a TODO here and will open an issue about when we merge this

@smarterclayton @ironcladlou